### ISSUE TITLE
Allow SVG upload (if SVG contains no scripting)

### ISSUE THREAD

**Issue by Zodiac1978** [WordPress/performance#357]
As mentioned [on Twitter](https://twitter.com/zodiac1978/status/1533024013068513280?s=20&t=F8yrZPueix_LJc6v-soQvA) after the talk from @felixarntz at WordCamp Europe 2022:

SVGs are much smaller in comparison to pixel based files like PNGs, JPGs or even webP files.

Using SVG has the advantage of small file sizes and additionally no problem with scaling because they are vector based.

Problem is, that SVGs can contain Javascript and therefore include a security risk.

@spacedmonkey had the idea of allowing just SVGs without any script in it (or block SVGs with scripting in it).

Related discussion can be found here on trac:
https://core.trac.wordpress.org/ticket/24251

There are 1 million+ active installs using this feature in some way:
https://de.wordpress.org/plugins/search/svg+upload/

Thanks for considering!

**Comment by masteradhoc** [comment]
Big +1 :)

**Comment by felixarntz** [comment]
Thanks @Zodiac1978, I think that's a great idea. +1 on not allowing any SVGs with scripts in it. cc @spacedmonkey 

I think we could advance this by implementing the functionality as a new module in the Performance Lab plugin? That way we can allow users to test this behavior in the wild before trying to get it into WordPress core without much feedback.

I'm not 100% sure WP core has the necessary integration points for checking whether the SVG contains scripts from within a plugin, but if not, maybe we can even add an action or filter to core as needed.

In other words, I would certainly support this effort. Any volunteers for implementing this as a module here? If so, we should also revamp this issue then to become a proper module proposal. See the following documentation:
* [Proposing a new module](https://github.com/WordPress/performance/blob/trunk/docs/Proposing-a-new-module.md)
* [Writing a module](https://github.com/WordPress/performance/blob/trunk/docs/Writing-a-module.md)

**Comment by spacedmonkey** [comment]
For what it worth, I have created similar module for the [web stories plugin](https://wordpress.org/plugins/web-stories/). It can be seen [here](https://github.com/GoogleForCreators/web-stories-wp/blob/main/includes/Media/SVG.php). 

This implemention uses [svg-sanitizer](https://github.com/darylldoyle/svg-sanitizer) from @darylldoyle to check if a SVG is clean. [See](https://github.com/GoogleForCreators/web-stories-wp/blob/98067edb59434eb4724f99e7ebb499942fbe6c65/includes/Media/SVG.php#L303-L326). This library would need to be vented, but it already used in [safe-svg](https://wordpress.org/plugins/safe-svg/). 

I am not sure what a benefit would be of creating a module on this plugin, when they are multiple plugins that exist here to test this idea.

**Comment by darylldoyle** [comment]
👋 Let me know if there's anything I can do to help here

**Comment by felixarntz** [comment]
@spacedmonkey 
> I am not sure what a benefit would be of creating a module on this plugin, when they are multiple plugins that exist here to test this idea.

The other plugins are not WordPress feature plugins. We need to allow users to test the feature/enhancement in isolation, in a way that it could also be integrated in WordPress core. With the implementation that is already in the Web Stories plugin, you could probably use that as inspiration for a module within the Performance Lab plugin.

**Comment by masteradhoc** [comment]
@Zodiac1978 @spacedmonkey @darylldoyle @felixarntz would love to get some feedback if the Module Proposal created in #410 (based on your and other existing discussions) is complete or you'd like me to integrate something? then i'd do it before the module gets discussed in slack :)

**Comment by mxbclang** [comment]
Thank you, @masteradhoc! Since we now have a separate module proposal issue for this with #410, I'm going to close this issue out so that we can continue discussion there.
